docs/dcp2_system_design.rst

```diff
@@ -935,6 +1028,413 @@ import tool with be requested manually in #ingest-to-tdr-shared Slack channel
+present in the TDR dataset. Under this definition, a deletion or removal marker
+object is considerd an altered copy. It is insignificant if the original was
+imported from a staging area at the same URI as the one containing the altered
+copy, or a different URI.
+
+We further differentiate between removals and deletions. Data files can be
+deleted or removed, subgraphs and metadata entities can only be removed. A
+removed data file, subgraph or metadata entity disappears from subsequently
+created snapshots but remains present in previously created snapshots. A deleted
+data file is both removed from subsequently created snapshots **and** erased
+physically from storage, references to it in previously created snapshots become
+invalid and requesting the data file's content results in an error.
+
+
+Update an entity
+~~~~~~~~~~~~~~~~
+
+Write the updated entity document to an object in the ``metadata`` directory of
+the staging area. The name of that object must have the same ``entity_type`` and
+``entity_id`` as the original and a strictly higher ``version``.
+
+
+Update a data file
+~~~~~~~~~~~~~~~~~~
+
+1. Write the updated data file to an object in the ``data`` directory of the
+   staging area, using the same relative path as the original data file.
+
+2. Write the updated descriptor to an object in the ``descriptors`` directory of
+   the staging area. The name of the descriptor object must have the same
+   ``entity_type`` and ``entity_id`` as the original, and a strictly higher
+   ``version``. The ``file_id`` property of the descriptor document must be the
+   same as that of the original, and the ``file_version`` property must be
+   strictly higher. The ``file_name`` property must have the same value as in
+   the original. Note that the ``file_id``property of the descriptor document is
+   independent from the column of the same name in TDR ``…_file`` tables and
+   that this section only applies to the former.
+
+   If the staging area has the `is_delta`` property set to ``true`` (or
+   ``false``), the ``sha1`` and ``sha256`` properties must (or should) differ
+   from their respective values in the orginal. [#]_ In other words, redundant
+   updates are not allowed in delta staging areas, and not recommended in
+   non-delta staging areas.
+
+   Similarly, if the ``size`` property is different, the ``sha1`` and ``sha256``
+   properties must differ between original and update. [#]_ The ``content-type``
+   property may have the same value or a different one. For updating *only* the
+   content type of a file see `Update the content type of a data file`_.
+
+3. Update the entity describing the data file, as specified in `Update an
+  entity`_. Do so even if the entity document itself did not change. This is
+  necessary so that the ``entity_type``, ``entity_id`` and ``version`` fields in
+  the name of the descriptor object match those in the name of the entity object.
```

Comment:
```suggestion
3. Update the entity describing the data file, as specified in `Update an
   entity`_. Do so even if the entity document itself did not change. This is
   necessary so that the ``entity_type``, ``entity_id`` and ``version`` fields in
   the name of the descriptor object match those in the name of the entity object.
```

Reply:
Good catch. Addressed.